### ISSUE TITLE
Grow graph as DAGNodes are loaded?

### ISSUE THREAD

**Issue by achingbrain** [ipfs-shipyard/ipfs-dag-builder-vis#11]
Loading big graphs is really slow because the UI waits for the whole thing to be loaded before rendering - could it add nodes to the graph as they come in instead?

**Comment by alanshaw** [comment]
We struggled with this initially, it could maybe look good when building it in traversal order i.e. after import, animate from the root node.

However, the DAG is actually created from bottom up, and the importer doesn't yield individual DAG nodes so there's a difference between what's happening in reality and what we'd be animating.

At the time we punted on this feature so we didn't confuse people, and it felt like a big time sink to get it working nicely. I've since added a loading spinner which should help with the UX a little.

**Comment by parkan** [comment]
I was able to use the tool successfully yesterday with a smaller file, but today I tried to upload a ~4MB MP3 a couple of times and it ballooned the tab to >1GiB memory usage and tons of CPU, and eventually froze (it seemed to be working rather hard after the "added" message in console so it sounds like the viz itself it pretty heavy?), now the site won't load at all

hopefully I didn't kill it 😬 

what I was attempting to determine was: would pre-loading the files ahead of time be quicker? seems like the answer might be "no" given that constructing the actual viz seems to be the heavy bit

**Comment by alanshaw** [comment]
If you're adding big files (bigger than say 100KB) then crank up the chunk size so there's fewer nodes to render in the graph.

There's no server component so you probably didn't break it! 😉

**Comment by parkan** [comment]
fair enough, 512b chunks yielded way too many nodes 😄 and the other downtime (?) seems to have been a coincidence
